baselines/flanders/flanders/strategy.py

```diff
@@ -0,0 +1,419 @@
+            None
+        to_keep : int, optional
+            Number of clients to keep (i.e., to classify as "good"), by default
+            1
+        window : int, optional
+            Sliding window size used as a "training set" of MAR, by default 0
+        maxiter : int, optional
+            Maximum number of iterations of MAR, by default 100
+        alpha : float, optional
+            Alpha parameter (regularization), by default 1
+        beta : float, optional
+            Beta parameter (regularization), by default 1
+        distance_function : Callable, optional
+            Distance function used to compute the distance between predicted
+            params and real ones, by default None
+        """
+        self.fraction_fit = fraction_fit
+        self.fraction_evaluate = fraction_evaluate
+        self.min_fit_clients = min_fit_clients
+        self.min_evaluate_clients = min_evaluate_clients
+        self.min_available_clients = min_available_clients
+        self.evaluate_fn = evaluate_fn
+        self.on_fit_config_fn = on_fit_config_fn
+        self.on_evaluate_config_fn = on_evaluate_config_fn
+        self.accept_failures = accept_failures
+        self.initial_parameters = initial_parameters
+        self.fit_metrics_aggregation_fn = fit_metrics_aggregation_fn
+        self.evaluate_metrics_aggregation_fn = evaluate_metrics_aggregation_fn
+        self.to_keep = to_keep
+        self.window = window
+        self.maxiter = maxiter
+        self.alpha = alpha
+        self.beta = beta
+        self.params_indexes = None
+        self.malicious_selected = False
+        self.distance_function = distance_function
+
+    def __repr__(self) -> str:
+        """Compute a string representation of the strategy."""
+        rep = f"FLANDERS(accept_failures={self.accept_failures})"
+        return rep
+
+    @typing.no_type_check
+    def num_fit_clients(self, num_available_clients: int) -> Tuple[int, int]:
+        """Return the sample size and the required number of available clients."""
+        num_clients = int(num_available_clients * self.fraction_fit)
+        return max(num_clients, self.min_fit_clients), self.min_available_clients
+
+    @typing.no_type_check
+    def num_evaluation_clients(self, num_available_clients: int) -> Tuple[int, int]:
+        """Use a fraction of available clients for evaluation."""
+        num_clients = int(num_available_clients * self.fraction_evaluate)
+        return max(num_clients, self.min_evaluate_clients), self.min_available_clients
+
+    # pylint: disable=unused-argument
+    @typing.no_type_check
+    def initialize_parameters(
+        self, client_manager: ClientManager
+    ) -> Optional[Parameters]:
+        """Initialize global model parameters."""
+        initial_parameters = self.initial_parameters
+        self.initial_parameters = None  # Don't keep initial parameters in memory
+        return initial_parameters
+
+    @typing.no_type_check
```

Comment:
I'm inclined to say It's better to try not to use this 😅. Are there some typing issues hard to solve in your implementation?

Reply:
It gives incompatible type errors, although it is code copied from FedAvg.

Reply:
ok. I'll work out something and propose the change.